### ISSUE TITLE
Unresolved Tempo hostname; unable to fetch in grafana for traceId

### ISSUE THREAD

**Issue by anil-toshniwal** [mnadeem/boot-opentelemetry-tempo#20]
Use branch: 0.15.0_basic
Stacktrace (when running docker compose up):

boot-otel-tempo-api_1        | [opentelemetry.auto.trace 2021-04-08 00:35:42:230 -0500] [grpc-default-executor-0] WARN io.opentelemetry.exporter.jaeger.JaegerGrpcSpanExporter - Failed to export spans
boot-otel-tempo-api_1        | io.grpc.StatusRuntimeException: UNAVAILABLE: Unable to resolve host tempo
boot-otel-tempo-api_1        | 	at io.grpc.Status.asRuntimeException(Status.java:533)
boot-otel-tempo-api_1        | 	at io.grpc.stub.ClientCalls$UnaryStreamToFuture.onClose(ClientCalls.java:533)
boot-otel-tempo-api_1        | 	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:617)
boot-otel-tempo-api_1        | 	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:70)
boot-otel-tempo-api_1        | 	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:803)
boot-otel-tempo-api_1        | 	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:782)
boot-otel-tempo-api_1        | 	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
boot-otel-tempo-api_1        | 	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
boot-otel-tempo-api_1        | 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
boot-otel-tempo-api_1        | 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
boot-otel-tempo-api_1        | 	at java.lang.Thread.run(Thread.java:748)
boot-otel-tempo-api_1        | Caused by: java.lang.RuntimeException: java.net.UnknownHostException: tempo: Name does not resolve
boot-otel-tempo-api_1        | 	at io.grpc.internal.DnsNameResolver.resolveAddresses(DnsNameResolver.java:223)
boot-otel-tempo-api_1        | 	at io.grpc.internal.DnsNameResolver.doResolve(DnsNameResolver.java:282)
boot-otel-tempo-api_1        | 	at io.grpc.internal.DnsNameResolver$Resolve.run(DnsNameResolver.java:318

**Comment by mnadeem** [comment]
Build the image

````
mvn clean package docker:build
````

run the images

````
docker-compose up
````

Things are fine

![image](https://user-images.githubusercontent.com/476145/114069461-a4bc2780-98bc-11eb-8a9d-ec7a890800f2.png)


![image](https://user-images.githubusercontent.com/476145/114069377-8fdf9400-98bc-11eb-8c37-fe5d577670e6.png)


![image](https://user-images.githubusercontent.com/476145/114069601-cc12f480-98bc-11eb-91dd-9cd15c98dd10.png)

**Comment by jetztgradnet** [comment]
I have the same issue (using branch `main` and running this with Docker Desktop for Mac).
The containers keep respawning.
```
boot-otel-tempo-provider1_1  | [BatchSpanProcessor_WorkerThread-1] WARN io.opentelemetry.exporter.jaeger.JaegerGrpcSpanExporter - Failed to export spans
boot-otel-tempo-provider1_1  | io.grpc.StatusRuntimeException: UNAVAILABLE: Unable to resolve host tempo
boot-otel-tempo-provider1_1  | 	at io.grpc.Status.asRuntimeException(Status.java:534)
boot-otel-tempo-provider1_1  | 	at io.grpc.stub.ClientCalls$UnaryStreamToFuture.onClose(ClientCalls.java:533)
boot-otel-tempo-provider1_1  | 	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:617)
boot-otel-tempo-provider1_1  | 	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:70)
boot-otel-tempo-provider1_1  | 	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:803)
boot-otel-tempo-provider1_1  | 	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:782)
boot-otel-tempo-provider1_1  | 	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
boot-otel-tempo-provider1_1  | 	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
boot-otel-tempo-provider1_1  | 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
boot-otel-tempo-provider1_1  | 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
boot-otel-tempo-provider1_1  | 	at java.lang.Thread.run(Thread.java:748)
boot-otel-tempo-provider1_1  | Caused by: java.lang.RuntimeException: java.net.UnknownHostException: tempo: Name does not resolve
boot-otel-tempo-provider1_1  | 	at io.grpc.internal.DnsNameResolver.resolveAddresses(DnsNameResolver.java:223)
boot-otel-tempo-provider1_1  | 	at io.grpc.internal.DnsNameResolver.doResolve(DnsNameResolver.java:282)
boot-otel-tempo-provider1_1  | 	at io.grpc.internal.DnsNameResolver$Resolve.run(DnsNameResolver.java:318)
boot-otel-tempo-provider1_1  | 	... 3 more
boot-otel-tempo-provider1_1  | Caused by: java.net.UnknownHostException: tempo: Name does not resolve
boot-otel-tempo-provider1_1  | 	at java.net.Inet4AddressImpl.lookupAllHostAddr(Native Method)
boot-otel-tempo-provider1_1  | 	at java.net.InetAddress$2.lookupAllHostAddr(InetAddress.java:929)
boot-otel-tempo-provider1_1  | 	at java.net.InetAddress.getAddressesFromNameService(InetAddress.java:1324)
boot-otel-tempo-provider1_1  | 	at java.net.InetAddress.getAllByName0(InetAddress.java:1277)
boot-otel-tempo-provider1_1  | 	at java.net.InetAddress.getAllByName(InetAddress.java:1193)
boot-otel-tempo-provider1_1  | 	at java.net.InetAddress.getAllByName(InetAddress.java:1127)
boot-otel-tempo-provider1_1  | 	at io.grpc.internal.DnsNameResolver$JdkAddressResolver.resolveAddress(DnsNameResolver.java:631)
boot-otel-tempo-provider1_1  | 	at io.grpc.internal.DnsNameResolver.resolveAddresses(DnsNameResolver.java:219)
boot-otel-tempo-provider1_1  | 	... 5 more
```

Do you maybe resolve the hostname `tempo` locally? Or use some specific Docker setup/configuration?

**Comment by jetztgradnet** [comment]
I also tried branch `0.17.0_complex` with same result. 
But then I notices these lines:
```
tempo_1                      | failed parsing config: failed to parse configFile /etc/tempo.yaml: yaml: unmarshal errors:
tempo_1                      |   line 38: field bloom_filter_false_positive not found in type wal.Config
tempo_1                      |   line 39: field index_downsample not found in type wal.Config
```
So apparently the `tempo` container is not coming up (it keeps respawning) because of an invalid config file.

**Comment by jetztgradnet** [comment]
Commenting the lines with `bloom_filter_false_positive` and `index_downsample` in `etc/tempo-local.yaml` fixes the issue.

The tempo image used in the docker-compose file is referring to the `latest` tag, so may you might have a different version which (still) supports those settings while the one I and @anil-toshniwal got might not?

Anyway, with the workaround of commenting those lines everything is coming up now. Thakns for a great integrated example!

**Comment by jetztgradnet** [comment]
Hm, this is still not fully working for me:
```
tempo_1                      | level=info ts=2021-05-06T23:19:36.3896858Z caller=tempodb.go:371 msg="compaction and retention enabled."
tempo_1                      | level=error ts=2021-05-06T23:19:36.4720902Z caller=frontend_processor.go:70 msg="error processing requests" address=127.0.0.1:9095 err="rpc error: code = Unavailable desc = transport is closing"
tempo_1                      | level=error ts=2021-05-06T23:19:36.4722082Z caller=frontend_processor.go:70 msg="error processing requests" address=127.0.0.1:9095 err="rpc error: code = Unavailable desc = transport is closing"

...

tempo-query_1                | {"level":"error","ts":1620343272.0990686,"caller":"zap/logger.go:33","msg":"failed to flush Jaeger spans to server: write udp 127.0.0.1:59407->127.0.0.1:6831: write: connection refused","stacktrace":"github.com/uber/jaeger-client-go/log/zap.(*Logger).Error\n\tgithub.com/uber/jaeger-client-go@v2.23.1+incompatible/log/zap/logger.go:33\ngithub.com/uber/jaeger-client-go.(*remoteReporter).processQueue.func1\n\tgithub.com/uber/jaeger-client-go@v2.23.1+incompatible/reporter.go:288\ngithub.com/uber/jaeger-client-go.(*remoteReporter).processQueue\n\tgithub.com/uber/jaeger-client-go@v2.23.1+incompatible/reporter.go:298"}
```
This looks like some ports (`6831`, which apparently is used by the Jaeger agent, and `9095`, opened by Loki) are not available. Is this just a matter of exposing them in the docker-compose file?

Any ideas or suggestions?

**Comment by mnadeem** [comment]
kindly check if those ports are already used in your machine

**Comment by mnadeem** [comment]
https://github.com/grafana/tempo/releases This would be helpful, I will update the images with correct release

**Comment by mnadeem** [comment]
Take care, bloom_filter_false_positive and index_downsample fixed the tempo version to 0.7.0

Thanks for opening this
